src/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -153,6 +151,12 @@ void smt2_incremental_decision_proceduret::define_dependent_functions(
     }
     if(const auto array_expr = expr_try_dynamic_cast<array_exprt>(current))
       define_array_function(*array_expr);
+    if (const auto nondet_symbol = expr_try_dynamic_cast<nondet_symbol_exprt>(current)) {
```

Comment:
Is our implementation of `nondet_symbol` just an uninterpreted function? Or do I misunderstand this?

Reply:
Yes, it's a fresh non-constrained uninterpreted function

Reply:
Yes. A `nondet_symbol_exprt` is equivalent to a `symbol_exprt` in the case where the symbolt has no defined value. It is just explicitly nondet rather than implicitly nondet as far as I can tell. Unless you know of any other differences Fotis? Symmex already seems to handle generating a new `nondet_symbol_exprt` identifier for each call to the undefined function, therefore I think we don't need to deal with generating new identifiers in the decision procedure.